File: bot/slack-processor-python/src/handlers/message.py
"""Message event handlers."""

from slack_bolt import App

from src.processors import get_supported_types


def register_message_handlers(app: App) -> None:
    """Register message-related handlers."""

    @app.event("app_mention")
    def handle_mention(event: dict, client, logger) -> None:
        """Respond to app mentions with help info."""
        try:
            supported = get_supported_types()
            client.chat_postMessage(
                channel=event["channel"],
                thread_ts=event.get("ts"),
                text=f"Hi! I'm a file processor bot. Upload a file and I'll process it.\n\n*Supported file types:*\n{supported}",
            )
        except Exception as e:
            logger.error(f"Error responding to mention: {e}")

    @app.message("help|도움|지원")
    def handle_help(message: dict, client, logger) -> None:
        """Respond to help requests."""
        try:
            supported = get_supported_types()
            client.chat_postMessage(
                channel=message["channel"],
                thread_ts=message.get("ts"),
                text=f"*Supported file types:*\n{supported}\n\nJust upload a file and I'll process it automatically!",
            )
        except Exception as e:
            logger.error(f"Error sending help: {e}")

File: bot/slack-processor-ts-node/src/handlers/message.ts
import type { App } from '@slack/bolt'
import { getSupportedTypes } from '../processors/index.js'

/**
 * Register message handlers
 *
 * Example handlers for app mentions and direct messages.
 * Customize these for your use case.
 */
export function registerMessageHandlers(app: App): void {
  // Respond to app mentions
  app.event('app_mention', async ({ event, client, logger }) => {
    try {
      const supportedTypes = getSupportedTypes()

      await client.chat.postMessage({
        channel: event.channel,
        thread_ts: event.ts,
        text: `Hi! I'm a file processor bot. Upload a file and I'll process it for you.\n\n*Supported file types:*\n${supportedTypes}`,
      })
    } catch (error) {
      logger.error('Error responding to mention:', error)
    }
  })

  // Example: React to specific keywords in messages
  app.message(/help|지원|도움/, async ({ message, client, logger }) => {
    try {
      // Type guard for message with channel
      if (!('channel' in message)) return

      const supportedTypes = getSupportedTypes()

      await client.chat.postMessage({
        channel: message.channel,
        thread_ts: 'ts' in message ? message.ts : undefined,
        text: `*Supported file types:*\n${supportedTypes}\n\nJust upload a file and I'll process it automatically!`,
      })
    } catch (error) {
      logger.error('Error sending help:', error)
    }
  })
}
